### ISSUE TITLE
Scheduler step location question

### ISSUE THREAD

**Issue by zhwj888** [ultralytics/yolov5#7485]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi, thanks for this great work.
I have a question about why in yolov5 the **scheduler.step** of OneCycleLR is put outside of the for loop of batch loading, which means the LR will be changed after every epoch.
![lr](https://user-images.githubusercontent.com/12584378/164065709-078d3740-7a45-4d2b-86f7-0e734accbc49.PNG)

But from pytorch doc, it indicates that **shceduler.step** of OneCycleLR should be put inside the for loop of batch loading, which means the LR will be changed after every minibatch.
![lr2](https://user-images.githubusercontent.com/12584378/164065743-2cb67d94-c193-495a-91f5-ffe5d7c4f2b1.PNG)

**Comment by glenn-jocher** [comment]
@zhwj888 hi, thank you for your feature suggestion on how to improve YOLOv5 🚀! I think it's mostly arbitrary, but if you'd like to try updating the YOLOv5 implementation to match the PyTorch docs that would make a great PR!

The fastest and easiest way to incorporate your ideas into the official codebase is to submit a Pull Request (PR) implementing your idea, and if applicable providing before and after profiling/inference/training results to help us understand the improvement your feature provides. This allows us to directly see the changes in the code and to understand how they affect workflows and performance.

Please see our ✅ [Contributing Guide](https://docs.ultralytics.com/help/contributing/) to get started.

**Comment by zhwj888** [comment]
@glenn-jocher Many thanks for your quick reply. 
I actually tried two positions. I took the yolov5 backbone as a classifier for my experiment project and training tricks too. 
So the conclution may not be applicable to yolov5: 
if the scheduler.step is put inside the for loop of batch loading the LR will drease slowly after every epoch and final shape can be bell shaped.
otherwise, the LR will rapidly changes from the highest to the lowest, then from the lowest to the highest and repeat the process. The final shape of the LR graph is a wavy line.

Based on the definition of OneCyleLR, the bell shaped looks correct. So I suppose the reason is the formula which is used in yolov5 is actully different with the one used in Pytorch official.

**Comment by glenn-jocher** [comment]
@zhwj888 well you're supposed to adjust the total iterations if you change the number obviously. You can do one complete cycle in any arbitrary number of steps, you just need to figure that out when you create the scheduler and pass the appropriate number so it compensates for if it's per batch or per epoch.

**Comment by zhwj888** [comment]
> @zhwj888 well you're supposed to adjust the total iterations if you change the number obviously. You can do one complete cycle in any arbitrary number of steps, you just need to figure that out when you create the scheduler and pass the appropriate number so it compensates for if it's per batch or per epoch.

Thanks, I got it
